File: script/benchmark/lang/prime/prime.py

import sys

def isprime(n, primes):
    sqrt = n ** 0.5
    for i in primes:
        if sqrt < i: break
        if n % i == 0: return False
    return True

def main():
    primes = []

    for i in range(2, int(sys.argv[1])):
        if isprime(i, primes):
            primes.append(i)
main()

File: script/benchmark/lang/prime/prime.rb
def isprime(n, primes)
    sqrt = n ** 0.5
    for i in primes
        break if sqrt < i
        return false if n % i == 0
    end
    return true
end

primes = []

for i in 2..ARGV[0].to_i
    primes.push(i) if isprime(i, primes)
end

